doc/sphinx/proof-engine/tactics.rst

```diff
@@ -460,16 +460,14 @@ Examples:
 Occurrence sets and occurrence clauses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-An occurrence clause is a modifier to some tactics that obeys the
+An occurrence clause (:n:`at @occurrences`) is a modifier to some tactics that obeys the
 following syntax:
 
-  .. productionlist:: coq
-     occurrence_clause : in `goal_occurrences`
-     goal_occurrences : [`ident` [`at_occurrences`], ... , `ident` [`at_occurrences`] [|- [* [`at_occurrences`]]]]
-                      : * |- [* [`at_occurrences`]]
-                      : *
-     at_occurrences : at `occurrences`
-     occurrences     : [-] `num` ... `num`
+  .. insertprodn occurrences occurrences
+
+  .. prodn::
+     occurrences ::= {+ @int }
+     | @ident
```

Comment:
Something is off with this grammar. As shown by the previous `productionlist`, the actual grammar is much more complex. Aren't you confusing `occurences` and the more complex `goal_occurences`?

Reply:
Nope.  The previous `productionlist` is incorrect.  `goal_occurrences` doesn't appear in the mlg files.  `occurrences` is:

```
occurrences: [
| LIST1 integer
| var
]
```

Reply:
`occurrence_clause` is `clause_dft_concl` or `clause_dft_all` or `opt_clause` in the mlg files.
`goal_occurrences` is `in_clause`.

Reply:
The term "occurrence clause" is confusing/overloaded .  I took it to refer to `"at" occurrences` in these tactics.  Then there's also `in_clause` which, through `clause_dft_concl` is used in 30 tactics.  And the `in_clause` also lets you select hypotheses.

```
| "rewrite" "*" orient term1_extended "in" ident "at" occurrences by_arg_tac
| "rewrite" "*" orient term1_extended "at" occurrences "in" ident by_arg_tac
| "rewrite" "*" orient term1_extended "at" occurrences by_arg_tac
| "setoid_rewrite" orient constr_with_bindings "at" occurrences
| "setoid_rewrite" orient constr_with_bindings "at" occurrences "in" ident
| "setoid_rewrite" orient constr_with_bindings "in" ident "at" occurrences
```

I substituted the `in_clause` grammar but the wording needs some tweaking.

Reply:
I agree we need a better name but I don't have any good suggestion.